tools/samtools/patches/1.21.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/Makefile b/Makefile
+index 671d838..be34fac 100644
+--- a/Makefile
++++ b/Makefile
+@@ -124,7 +124,7 @@ version.h:
+ 	echo '#define SAMTOOLS_CC "$(CC)"' >> $@
+ 	echo '#define SAMTOOLS_CPPFLAGS "$(CPPFLAGS)"' >> $@
+ 	echo '#define SAMTOOLS_CFLAGS "$(CFLAGS)"' >> $@
+-	echo '#define SAMTOOLS_LDFLAGS "$(LDFLAGS)"' >> $@
++	echo '#define SAMTOOLS_LDFLAGS "--preload-file examples/@/samtools/examples/ -s ERROR_ON_UNDEFINED_SYMBOLS=0 -O2"' >> $@
+ 	echo '#define SAMTOOLS_HTSDIR "$(HTSDIR)"' >> $@
+ 	echo '#define SAMTOOLS_LIBS "$(LIBS)"' >> $@
+ 	echo '#define SAMTOOLS_CURSES_LIB "$(CURSES_LIB)"' >> $@
+@@ -142,7 +142,7 @@ LIBST_OBJS = sam_opts.o sam_utils.o bedidx.o bam.o
+ 
+ 
+ samtools: $(AOBJS) $(LZ4OBJS) libst.a $(HTSLIB)
+-	$(CC) $(ALL_LDFLAGS) -o $@ $(AOBJS) $(LZ4OBJS) libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
++	$(CC) $(ALL_LDFLAGS) -o ../build/$@.js $(AOBJS) $(LZ4OBJS) libst.a $(HTSLIB_LIB) $(CURSES_LIB) -lm $(ALL_LIBS) -lpthread
+ 
+ # For building samtools and its test suite only: NOT to be installed.
+ libst.a: $(LIBST_OBJS)
+diff --git a/bam_sort.c b/bam_sort.c
+index 2ff6f7e..ca48903 100644
+--- a/bam_sort.c
++++ b/bam_sort.c
+@@ -3119,15 +3119,6 @@ static int sort_blocks(size_t k, bam1_tag *buf, const sam_hdr_t *h,
+         in_mem[i].from = pos;
+         in_mem[i].to = pos + w[i].buf_len;
+         pos += w[i].buf_len; rest -= w[i].buf_len;
+-        pthread_create(&tid[i], &attr, worker, &w[i]);
+-    }
+-    for (i = 0; i < n_threads; ++i) {
+-        pthread_join(tid[i], 0);
+-        if (w[i].error != 0) {
+-            errno = w[i].error;
+-            print_error_errno("sort", "failed to sort block %d", i);
+-            n_failed++;
+-        }
+     }
+     free(w);
```

Comment:
I think you'll need to call the function directly if you disable threads:

```suggestion
     worker(&w[0]);
     free(w);
```

Reply:
Sounds absolutely right! I'll cowardly not commit from your suggestion back to the fork, but I will be sure to use it!

Reply:
Oups, looks like that didn't work, I get `error: corrupt patch` because we're missing a `+` in front of the `worker(&w[0]);` line. Or might be easier to regenerate with `diff`?

Reply:
Oopsie indeed. I'll re-diff to be sure. 

You see, I should have just accepted the suggestion directly and worked with git. Chickening out seldom pays. 😆

Reply:
Nah, my suggestion was also missing the + sign 😂

Reply:
To our defence it was a little difficult to spot, with the GitHub git diff `+` in front of where the script git diff `+` should be. 🕵️